src/App.tsx

```diff
@@ -1,61 +1,102 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './types/totos';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [selectUserError, setSelectUserError] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [allTodos, setAllTodos] = useState<Todo[]>(todos);
+
+  const reset = () => {
+    setTitle('');
+    setSelectUser(0);
+    setSelectUserError('');
+    setTitleError('');
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError('title');
+    }
+
+    if (!selectUser) {
+      setSelectUserError('user');
+    }
+
+    if (title && selectUser) {
+      const newTodo: Todo = {
+        id: Math.max(...todos.map(todo => todo.id)) + 1,
+        title,
+        completed: false,
+        userId: +selectUser,
+        user: usersFromServer.find(user => +selectUser === user.id) || null,
```

Comment:
It is better to move this function out of the component for better readability.
Example: 
```
function getUserById(userId) {
  return usersFromServer.find(user => user.id === userId) || null;
}
```

```suggestion
        user: getUserById(+selectUser),
```